cmake/download_test_data.cmake

```diff
@@ -0,0 +1,11 @@
+find_package(Git)
+
+# target to download test data
+add_custom_target(download_test_data
+    COMMAND test -d json_test_data || ${GIT_EXECUTABLE} clone https://github.com/nlohmann/json_test_data.git --quiet --depth 1
```

Comment:
Shouldn't we pin down the version of `json_test_data` we are referring to? Otherwise we have to worry about backward-compatibility of changes in `json_test_data` in future.

Reply:
Good catch! I would use semantic versioning for the data repo, rightß

Reply:
Can versioning it as something like v1, v2, v3 do the job? There is no harm in using semantic versioning, but it _might_ be overkill in this case since we are not providing an API.

Reply:
I totally agree. Will do.

Reply:
I still decided for semantic versioning - you'll never know ;)